# Estatística Aplicada I
## Segunda Lista de Exercícios

#### Estudante: Clístenes Grizafis Bento

### Com a base de dados “imoveiscwbav” obter os seguintes resultados com o auxílio do “R”

### a)	Estimar um modelo preliminar e apresentar os resultados;

In [197]:
#Carreando bibliotecas
load("imoveiscwbav.RData")
gc()

used    (Mb)  gc trigger (Mb)  max used (Mb) 
Ncells 3712662 198.3  7625532   407.3  7625532 407.3
Vcells 6892518  52.6 12267951    93.6 10759652  82.1

In [198]:
resultados <- lm(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                   trans+kidca+school+health+bike+barb+balc+elev+
                   fitg+party+categ,data=imoveiscwbav)


summary (resultados)


Call:
lm(formula = price ~ age + parea + tarea + bath + ensuit + garag + 
    plaz + park + trans + kidca + school + health + bike + barb + 
    balc + elev + fitg + party + categ, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-495718 -134211   -2632  104528 2419265 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -420453.5   130052.5  -3.233   0.0013 ** 
age           -7839.1     1025.3  -7.645 1.01e-13 ***
parea          2592.2      624.0   4.154 3.82e-05 ***
tarea          1975.8      333.9   5.918 5.91e-09 ***
bath          13452.6    14832.9   0.907   0.3649    
ensuit       125949.6    18560.7   6.786 3.15e-11 ***
garag        169687.5    21756.1   7.800 3.41e-14 ***
plaz         224393.0    94219.1   2.382   0.0176 *  
park         -63439.6    27154.0  -2.336   0.0199 *  
trans         26642.3    22718.5   1.173   0.2414    
kidca         10452.8    34899.8   0.300   0.7647    
school        -7975.8    56635.7  -0.141

In [199]:
imoveiscwbav

price   age parea tarea bath ensuit garag plaz       park      trans    
1   1100000 15  150   190   4    1      2     0.08058169 0.7132806 2.3862709
2    895000 11  165   210   4    1      2     0.16635098 0.6983694 2.2463043
3   2513600  2  146   275   4    3      3     0.05607530 1.3129824 2.6314112
4    755000 25  163   238   3    1      2     0.32159391 2.1099578 2.1387003
5   1099000  1  107   189   3    1      2     0.14663511 1.0175299 1.7978931
6    475000 31   96   124   2    1      1     0.12615199 1.9700244 0.9947105
7    463900  2   75    90   2    1      1     0.22649932 2.6027009 1.4633611
8   1320000  1  122   227   3    1      2     0.19124408 1.1613287 2.8183332
9    550000 11   63    87   2    1      1     0.46403846 1.9965809 1.8184778
10   831000  1   97   180   2    1      2     0.19130703 2.1665835 2.3593991
11   979000  3   92   130   2    1      3     0.35417661 2.0669094 2.2013961
12  1817417  3  138   253   3    3      2     0.39193020 2.1114371 1.7348651
13  2892000  3  199   405   5    3      3     0.11616112 0.7025960 2.4255204
14  1050000 11  150   255   3    1      2     0.08058169 0.7132806 2.3862709
15   959000  3   92   130   2    1      2     0.35417661 2.0669094 2.2013961
16   490000 20   86   148   3    1      1     0.12022662 2.1309628 1.8639108
17  1548000  3  139   241   5    3      2     0.35713365 2.3320754 1.4810736
18  2835000  3  199   284   5    3      3     0.11587989 0.7088692 2.4374977
19   850000 38  184   281   2    1      2     0.30068165 2.3396719 1.4572900
20   995000 16  164   206   4    3      2     0.07345642 1.4411875 2.5190436
21  2017000  3  150   215   5    3      3     0.11587989 0.7088692 2.4374977
22  1692000  3  139   241   4    3      3     0.35713365 2.3320754 1.4810736
23  1720000  3  150   215   5    3      3     0.11587989 0.7088692 2.4374977
24   530000 22   78    97   2    1      1     0.21268581 0.8530269 2.3086127
25  1850000  1  158   297   5    2      3     0.34226296 2.0804271 2.2631276
26  1079000  1  107   196   3    1      2     0.14663511 1.0175299 1.7978931
27  1485603  3  155   241   3    3      2     0.39193020 2.1114371 1.7348651
28   979000 16  145   215   4    1      2     0.08318506 0.8471284 2.4272793
29  1358000  3  121   185   3    1      2     0.17224844 2.2469464 2.1452308
30   730000 13  110   128   2    1      2     0.23733018 1.9439387 0.7801857
... ...     ... ...   ...   ...  ...    ...   ...        ...       ...      
512 1864000  3  155   254   4    3      2     0.3559855  2.331109  1.481412 
513 1494000  3  139   241   4    3      2     0.3559855  2.331109  1.481412 
514 1850000  3  158   258   5    3      3     0.3370544  2.084975  2.256744 
515 1320000  1  122   228   3    1      2     0.1974011  1.160932  2.814845 
516 1920000  3  132   228   4    2      3     0.1449806  1.419008  2.688781 
517 1180000  1  122   225   3    1      2     0.1936256  1.160322  2.818933 
518  340000 34   79   115   2    1      1     0.1361207  2.531448  1.944107 
519  750000 12  162   213   3    1      3     0.1519592  2.216822  1.376249 
520  520000 10  135   190   2    1      1     0.4983964  1.874185  1.952006 
521  595000 17  131   180   3    1      2     0.2233063  2.396326  1.234467 
522  740000 12  162   259   3    1      3     0.3173683  2.235832  1.541227 
523  949000  1  147   217   4    3      2     0.2328303  2.101013  2.413553 
524  820000 10  140   155   3    1      2     0.4354084  1.984615  2.299085 
525  858980  1  148   200   2    3      2     0.0835254  2.190340  1.925189 
526  950000  1  179   188   5    3      2     0.2485592  2.453074  2.529130 
527  950013  1  179   188   4    3      2     0.2485592  2.453074  2.529130 
528  950000  1  179   188   5    3      2     0.2485592  2.453074  2.529130 
529  950000  1  179   188   3    3      2     0.2485592  2.453074  2.529130 
530  950040  1  179   188   4    3      2     0.2485592  2.453074  2.529130 
531  950001  1  179   188   4    3      2     0.2485592  2.453074  2.529130 


### b) Testar as variáveis para formulação do modelo;

In [200]:
library("PanJen")

#teste com age
formBase<-formula(price~parea+tarea+bath+ensuit+garag+plaz+park+
                 trans+kidca+school+health+bike+barb+balc+elev+
                 fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenage<-fform(imoveiscwbav,"age",formBase)



Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-559997 -135850   -7282  108797 2360349 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -593909.9   134921.3  -4.402 1.30e-05 ***
parea          1643.3      644.3   2.550 0.011042 *  
tarea          2396.9      346.9   6.909 1.43e-11 ***
bath          -8495.5    15332.4  -0.554 0.579756    
ensuit       172325.9    18481.5   9.324  < 2e-16 ***
garag        215693.1    22027.9   9.792  < 2e-16 ***
plaz         250882.7    99201.5   2.529 0.011732 *  
park         -51641.8    28563.1  -1.808 0.071183 .  
trans         36393.6    23898.3   1.523 0.128401    
kidca          4784.5    36762.0   0.130 0.896500    
school        -2905.4    59667.0  -0.049 0.961182    
health        35316.9    59042.6   0.598 0.549993    
bike        -132282.7    58730.9  -2.252 0.024715 *  
barb         -21661.4    23615.2  -0.917 0.359427    
balc          91190.0    26351

               AIC      BIC ranking (BIC)
smoothing 14879.74 14977.41             1
x^2       14897.14 14987.30             2
log(x)    14901.06 14991.22             3
x         14912.25 15002.42             4
x+x^2     14943.68 15033.85             5
sqr(x)    14944.35 15034.51             6
1/x       14957.80 15047.96             7
base      14967.78 15053.65             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


##### smoothing  vencedora, incluir age

In [201]:
formBase<-formula(price~age+tarea+bath+ensuit+garag+plaz+park+
                    trans+kidca+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenparea<-fform(imoveiscwbav,"parea",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-536220 -128848    1494  108580 2420400 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -345157.5   130773.1  -2.639  0.00855 ** 
age           -6991.9     1020.4  -6.852 2.06e-11 ***
tarea          2853.5      262.5  10.870  < 2e-16 ***
bath          23570.7    14857.7   1.586  0.11325    
ensuit       139534.2    18552.6   7.521 2.39e-13 ***
garag        180717.9    21927.1   8.242 1.38e-15 ***
plaz         225240.4    95674.8   2.354  0.01893 *  
park         -56223.2    27517.1  -2.043  0.04153 *  
trans         41171.8    22794.5   1.806  0.07146 .  
kidca          9002.4    35437.3   0.254  0.79957    
school        23779.1    56984.6   0.417  0.67664    
health         3818.0    57081.6   0.067  0.94670    
bike        -105147.8    56744.0  -1.853  0.06444 .  
barb         -41841.0    22945.9  -1.823  0.06881 .  
balc          67611.1    25625

               AIC      BIC ranking (BIC)
sqr(x)    14904.77 14994.94             1
x+x^2     14904.80 14994.96             2
x         14912.25 15002.42             3
smoothing 14881.96 15002.43             4
x^2       14916.70 15006.87             5
log(x)    14921.03 15011.19             6
base      14927.88 15013.75             7
1/x       14929.78 15019.94             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### sqr(x) vencedora, incluir parea         

In [202]:
library(timetk)

formBase<-formula(price~age+parea+bath+ensuit+garag+plaz+park+
                    trans+kidca+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJentarea<-fform(imoveiscwbav,"tarea",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-473778 -125283   -4964   95015 2459849 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -425254.9   134221.1  -3.168  0.00162 ** 
age           -8840.2     1043.7  -8.470 2.51e-16 ***
parea          4929.2      498.7   9.885  < 2e-16 ***
bath          22083.6    15234.5   1.450  0.14778    
ensuit       122215.9    19144.9   6.384 3.82e-10 ***
garag        195599.8    21994.5   8.893  < 2e-16 ***
plaz         215182.2    97227.8   2.213  0.02732 *  
park         -75332.0    27948.1  -2.695  0.00726 ** 
trans         19109.4    23410.3   0.816  0.41471    
kidca         26619.7    35908.7   0.741  0.45884    
school       -24685.9    58379.6  -0.423  0.67258    
health        -7441.8    57999.9  -0.128  0.89796    
bike         -74259.7    57836.1  -1.284  0.19972    
barb         -44110.9    23327.2  -1.891  0.05918 .  
balc          62890.8    26049

               AIC      BIC ranking (BIC)
smoothing 14863.46 14963.98             1
sqr(x)    14886.91 14977.07             2
x+x^2     14886.96 14977.12             3
x         14912.25 15002.42             4
x^2       14925.17 15015.33             5
log(x)    14935.80 15025.96             6
base      14945.45 15031.32             7
1/x       14947.12 15037.28             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### smoothing vencedora, incluir tarea       

In [203]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+park+
                    trans+kidca+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenplaz<-fform(imoveiscwbav,"plaz",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-487535 -137496   -4432  108110 2436449 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -445551.8   130203.6  -3.422  0.00067 ***
age           -7928.9     1029.2  -7.704 6.69e-14 ***
parea          2595.4      626.8   4.141 4.04e-05 ***
tarea          1962.7      335.3   5.853 8.53e-09 ***
bath          14915.5    14886.4   1.002  0.31683    
ensuit       123964.4    18624.7   6.656 7.14e-11 ***
garag        170152.7    21852.4   7.786 3.73e-14 ***
park         -42810.2    25850.3  -1.656  0.09831 .  
trans         29699.8    22783.5   1.304  0.19296    
kidca          5371.1    34990.1   0.154  0.87806    
school        32969.5    54204.2   0.608  0.54329    
health         4238.3    56453.1   0.075  0.94018    
bike         -67380.9    55781.3  -1.208  0.22761    
barb         -49106.4    22597.8  -2.173  0.03022 *  
balc          67618.0    25333

               AIC      BIC ranking (BIC)
log(x)    14910.84 15001.00             1
x^2       14911.34 15001.50             2
base      14916.11 15001.98             3
x         14912.25 15002.42             4
x+x^2     14912.90 15003.06             5
sqr(x)    14914.25 15004.41             6
1/x       14915.88 15006.04             7
smoothing 14904.11 15015.66             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### base vencedora, excluir plaz  

In [204]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+
                    trans+kidca+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenpark<-fform(imoveiscwbav,"park",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-485013 -143277  -12614  107799 2401659 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -619461.9    98692.8  -6.277 7.28e-10 ***
age           -7703.0     1028.1  -7.493 2.90e-13 ***
parea          2498.9      625.4   3.996 7.37e-05 ***
tarea          2033.5      334.4   6.082 2.30e-09 ***
bath          15777.2    14862.6   1.062  0.28894    
ensuit       121628.3    18547.0   6.558 1.31e-10 ***
garag        173458.1    21788.6   7.961 1.07e-14 ***
plaz         154175.0    89677.3   1.719  0.08617 .  
trans         52060.7    20028.9   2.599  0.00961 ** 
kidca         11610.9    35045.0   0.331  0.74054    
school        55960.3    49796.4   1.124  0.26162    
health       -10386.0    56235.3  -0.185  0.85354    
bike        -133014.7    52538.1  -2.532  0.01164 *  
barb         -42641.7    22691.9  -1.879  0.06078 .  
balc          70770.2    25234

               AIC      BIC ranking (BIC)
1/x       14905.25 14995.41             1
smoothing 14895.01 14995.66             2
log(x)    14906.77 14996.93             3
x^2       14909.76 14999.92             4
base      14915.89 15001.76             5
x         14912.25 15002.42             6
x+x^2     14914.73 15004.89             7
sqr(x)    14915.32 15005.48             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### smoothing  vencedora, incluir park

In [205]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                    kidca+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJentrans<-fform(imoveiscwbav,"trans",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-499136 -132980   -5716  104035 2418690 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -357200.0   118382.6  -3.017  0.00267 ** 
age           -7906.6     1024.1  -7.721 5.95e-14 ***
parea          2704.9      616.8   4.385 1.40e-05 ***
tarea          1953.9      333.5   5.859 8.24e-09 ***
bath          13235.2    14837.1   0.892  0.37279    
ensuit       124875.6    18544.7   6.734 4.37e-11 ***
garag        170325.1    21757.1   7.828 2.77e-14 ***
plaz         230636.8    94102.4   2.451  0.01458 *  
park         -78689.5    23846.3  -3.300  0.00103 ** 
kidca         25866.4    32341.9   0.800  0.42420    
school       -21418.6    55483.5  -0.386  0.69963    
health        10519.7    55674.0   0.189  0.85020    
bike         -77781.5    55667.8  -1.397  0.16293    
barb         -46944.8    22463.3  -2.090  0.03711 *  
balc          64635.0    25247

               AIC      BIC ranking (BIC)
1/x       14906.86 14997.02             1
base      14911.68 14997.55             2
log(x)    14910.50 15000.66             3
x^2       14911.49 15001.65             4
smoothing 14901.64 15001.97             5
x         14912.25 15002.42             6
x+x^2     14912.98 15003.14             7
sqr(x)    14913.14 15003.31             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### base vencedora, excluir trans

In [206]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                    trans+school+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenkidca<-fform(imoveiscwbav,"kidca",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-502179 -133678   -1484  105125 2419604 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -414146.4   128224.3  -3.230  0.00132 ** 
age           -7832.6     1024.2  -7.647 9.94e-14 ***
parea          2590.3      623.4   4.155 3.80e-05 ***
tarea          1983.6      332.6   5.965 4.52e-09 ***
bath          13727.7    14791.5   0.928  0.35380    
ensuit       126023.1    18542.8   6.796 2.94e-11 ***
garag        169361.9    21710.0   7.801 3.36e-14 ***
plaz         222667.7    93960.8   2.370  0.01816 *  
park         -63555.1    27127.6  -2.343  0.01951 *  
trans         29204.9    21027.5   1.389  0.16546    
school        -8130.1    56584.0  -0.144  0.88581    
health        -3562.0    53857.1  -0.066  0.94729    
bike         -82185.7    54663.6  -1.503  0.13332    
barb         -43846.3    22581.0  -1.942  0.05271 .  
balc          64650.0    25166

               AIC      BIC ranking (BIC)
base      14910.35 14996.22             1
log(x)    14912.07 15002.23             2
x^2       14912.13 15002.29             3
x         14912.25 15002.42             4
sqr(x)    14912.32 15002.48             5
1/x       14912.34 15002.50             6
x+x^2     14912.35 15002.51             7
smoothing 14887.98 15005.85             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### base vencedora, excluir kidca

In [207]:
ormBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                    trans+kidca+health+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenschool<-fform(imoveiscwbav,"school",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-502179 -133678   -1484  105125 2419604 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -414146.4   128224.3  -3.230  0.00132 ** 
age           -7832.6     1024.2  -7.647 9.94e-14 ***
parea          2590.3      623.4   4.155 3.80e-05 ***
tarea          1983.6      332.6   5.965 4.52e-09 ***
bath          13727.7    14791.5   0.928  0.35380    
ensuit       126023.1    18542.8   6.796 2.94e-11 ***
garag        169361.9    21710.0   7.801 3.36e-14 ***
plaz         222667.7    93960.8   2.370  0.01816 *  
park         -63555.1    27127.6  -2.343  0.01951 *  
trans         29204.9    21027.5   1.389  0.16546    
school        -8130.1    56584.0  -0.144  0.88581    
health        -3562.0    53857.1  -0.066  0.94729    
bike         -82185.7    54663.6  -1.503  0.13332    
barb         -43846.3    22581.0  -1.942  0.05271 .  
balc          64650.0    25166

               AIC      BIC ranking (BIC)
base      14910.35 14996.22           1.5
x         14910.35 14996.22           1.5
1/x       14908.25 14998.41           3.0
log(x)    14911.83 15001.99           4.0
x^2       14912.08 15002.24           5.0
sqr(x)    14912.25 15002.42           6.5
x+x^2     14912.25 15002.42           6.5
smoothing 14900.87 15017.99           8.0
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"
[1] "please note that you included school in the base-formula and it is also the variable you test"


#### base vencedora, excluir school

In [208]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                    trans+kidca+school+bike+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenhealth<-fform(imoveiscwbav,"health",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-495906 -134106   -2849  104632 2419065 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -419586.4   123616.3  -3.394  0.00074 ***
age           -7840.9     1021.1  -7.679 7.99e-14 ***
parea          2592.4      623.4   4.159 3.74e-05 ***
tarea          1975.6      333.4   5.925 5.67e-09 ***
bath          13448.6    14817.6   0.908  0.36450    
ensuit       125913.1    18466.1   6.819 2.55e-11 ***
garag        169707.3    21716.0   7.815 3.05e-14 ***
plaz         224439.0    94104.9   2.385  0.01744 *  
park         -63387.7    27021.9  -2.346  0.01936 *  
trans         26711.7    22469.7   1.189  0.23506    
kidca         10238.3    33432.2   0.306  0.75954    
school        -8287.5    54724.4  -0.151  0.87969    
bike         -85972.2    55798.1  -1.541  0.12398    
barb         -43910.8    22570.1  -1.946  0.05225 .  
balc          65104.2    25148

               AIC      BIC ranking (BIC)
base      14910.26 14996.12           1.0
1/x       14911.26 15001.42           2.0
sqr(x)    14912.08 15002.24           3.0
x+x^2     14912.21 15002.37           4.5
log(x)    14912.21 15002.37           4.5
x^2       14912.23 15002.39           6.0
x         14912.25 15002.42           7.0
smoothing 14911.60 15004.47           8.0
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### base vencedora, excluir health

In [209]:
formBase<-formula(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                    trans+kidca+school+health+barb+balc+elev+
                    fitg+party+categ)

summary(lm(formBase, data=imoveiscwbav))

PanJenbike<-fform(imoveiscwbav,"bike",formBase)


Call:
lm(formula = formBase, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-510703 -134525      33  104312 2420136 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -399412.6   129491.1  -3.084  0.00215 ** 
age           -8009.1     1020.6  -7.847 2.42e-14 ***
parea          2671.3      622.7   4.290 2.13e-05 ***
tarea          1957.9      334.1   5.860 8.19e-09 ***
bath          14345.4    14840.5   0.967  0.33418    
ensuit       126035.0    18584.5   6.782 3.22e-11 ***
garag        168530.2    21771.0   7.741 5.15e-14 ***
plaz         204424.0    93432.4   2.188  0.02912 *  
park         -78405.3    25366.9  -3.091  0.00210 ** 
trans         22366.0    22575.3   0.991  0.32228    
kidca         -1253.1    34096.2  -0.037  0.97070    
school       -23111.0    55838.4  -0.414  0.67912    
health         8860.3    56066.6   0.158  0.87449    
barb         -50094.9    22269.0  -2.250  0.02489 *  
balc          64100.9    25265

               AIC      BIC ranking (BIC)
base      14912.68 14998.55             1
log(x)    14910.52 15000.68             2
x^2       14911.06 15001.22             3
x         14912.25 15002.42             4
x+x^2     14912.97 15003.13             5
sqr(x)    14913.65 15003.82             6
1/x       14914.12 15004.28             7
smoothing 14909.28 15007.97             8
[1] "Smoothing is a semi-parametric and data-driven transformation, please see Wood (2006) for an elaboration"


#### base vencedora, excluir bike

In [210]:
resultados <- lm(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
                   trans+kidca+school+health+bike+barb+balc+elev+
                   fitg+party+categ,data=imoveiscwbav)
summary (resultados)


Call:
lm(formula = price ~ age + parea + tarea + bath + ensuit + garag + 
    plaz + park + trans + kidca + school + health + bike + barb + 
    balc + elev + fitg + party + categ, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-495718 -134211   -2632  104528 2419265 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -420453.5   130052.5  -3.233   0.0013 ** 
age           -7839.1     1025.3  -7.645 1.01e-13 ***
parea          2592.2      624.0   4.154 3.82e-05 ***
tarea          1975.8      333.9   5.918 5.91e-09 ***
bath          13452.6    14832.9   0.907   0.3649    
ensuit       125949.6    18560.7   6.786 3.15e-11 ***
garag        169687.5    21756.1   7.800 3.41e-14 ***
plaz         224393.0    94219.1   2.382   0.0176 *  
park         -63439.6    27154.0  -2.336   0.0199 *  
trans         26642.3    22718.5   1.173   0.2414    
kidca         10452.8    34899.8   0.300   0.7647    
school        -7975.8    56635.7  -0.141

### c)	Verifique a presença de outliers;


In [211]:
library (carData)
library(car)

outlierTest(resultados)

     rstudent unadjusted p-value Bonferroni p
393 12.131725         5.3325e-30   2.8849e-27
13   4.302072         2.0218e-05   1.0938e-02

### d)	Teste a especificação do modelo e altere se necessário;

In [212]:
library (zoo)
library (lmtest)

resettest(price~age+parea+tarea+bath+ensuit+garag+plaz+park+
            trans+kidca+school+health+bike+barb+balc+elev+
            fitg+party+categ,power=2:3,type="regressor", data=imoveiscwbav)


	RESET test

data:  price ~ age + parea + tarea + bath + ensuit + garag + plaz +     park + trans + kidca + school + health + bike + barb + balc +     elev + fitg + party + categ
RESET = 4.3391, df1 = 38, df2 = 483, p-value = 9.761e-15


#### F tabelado

In [213]:
qf(0.95, df1=38, df2=483)

[1] 1.429987

##### Como o F calculado (4.3391) é maior que o F tabelado (1,4299873657331), existe erro de especificação do modelo

Deixaremos para remover as variáveis posteriormente

### e)	Teste a presença de multicolinearidade e exclua variáveis se necessário;


In [214]:
cor(imoveiscwbav[,c("age", "parea", "tarea", "bath", "ensuit", "garag", "plaz", "park",
                      "trans", "kidca", "school", "health", "bike")], 
    use="complete")

library(car)

vif(lm(price~age+parea+tarea+bath+ensuit+garag+plaz+park+trans+kidca+school+health+bike,data=imoveiscwbav), type="high-order")

age         parea       tarea       bath        ensuit      garag       
age     1.00000000 -0.23046283 -0.33295431 -0.26832783 -0.44532055 -0.473613016
parea  -0.23046283  1.00000000  0.81435888  0.67748403  0.59932048  0.592953262
tarea  -0.33295431  0.81435888  1.00000000  0.64462760  0.56271720  0.639279714
bath   -0.26832783  0.67748403  0.64462760  1.00000000  0.73616242  0.574197743
ensuit -0.44532055  0.59932048  0.56271720  0.73616242  1.00000000  0.531151588
garag  -0.47361302  0.59295326  0.63927971  0.57419774  0.53115159  1.000000000
plaz    0.02522293 -0.06585725 -0.09211596 -0.06282817 -0.04448629 -0.079925081
park   -0.03471480 -0.21091786 -0.23195234 -0.16996914 -0.02382108 -0.162058611
trans  -0.01343249  0.28699371  0.23256402  0.16864749  0.06323091  0.170694456
kidca   0.01027489  0.23650295  0.25872055  0.23256465  0.15681143  0.131592614
school  0.08473405  0.16119021  0.09423370  0.07286173  0.02216520  0.005902974
health -0.08768266 -0.13449226 -0.13228191 -0.14426728 -0.12274559 -0.022402611
bike    0.07658592 -0.07005757 -0.07144375 -0.06967850 -0.05244026 -0.068895267
       plaz        park        trans       kidca       school       health     
age     0.02522293 -0.03471480 -0.01343249  0.01027489  0.084734049 -0.08768266
parea  -0.06585725 -0.21091786  0.28699371  0.23650295  0.161190207 -0.13449226
tarea  -0.09211596 -0.23195234  0.23256402  0.25872055  0.094233700 -0.13228191
bath   -0.06282817 -0.16996914  0.16864749  0.23256465  0.072861730 -0.14426728
ensuit -0.04448629 -0.02382108  0.06323091  0.15681143  0.022165200 -0.12274559
garag  -0.07992508 -0.16205861  0.17069446  0.13159261  0.005902974 -0.02240261
plaz    1.00000000  0.31662772 -0.15779668 -0.08194894  0.184979230 -0.01211491
park    0.31662772  1.00000000 -0.50306548 -0.27488761 -0.378827033  0.22422407
trans  -0.15779668 -0.50306548  1.00000000  0.44121318  0.082776136 -0.06120773
kidca  -0.08194894 -0.27488761  0.44121318  1.00000000  0.146596839 -0.34925210
school  0.18497923 -0.37882703  0.08277614  0.14659684  1.000000000 -0.37352482
health -0.01211491  0.22422407 -0.06120773 -0.34925210 -0.373524820  1.00000000
bike    0.27145206  0.32559215 -0.02767855  0.18358136  0.070783858 -0.11611012
       bike       
age     0.07658592
parea  -0.07005757
tarea  -0.07144375
bath   -0.06967850
ensuit -0.05244026
garag  -0.06889527
plaz    0.27145206
park    0.32559215
trans  -0.02767855
kidca   0.18358136
school  0.07078386
health -0.11611012
bike    1.00000000

age    parea    tarea     bath   ensuit    garag     plaz     park 
1.548520 3.864017 3.606353 2.955296 2.773746 2.159104 1.293432 2.321960 
   trans    kidca   school   health     bike 
1.755870 1.554352 1.592615 1.365034 1.359943

##### Parece que o parea e tarea são correlacionadas pela matriz de correlação. Pelo valor de inflação de variancia nota-se que que o parea e o tarea tem score alto, quase atingindo valor quatro, então estão consideravelmente correlacionados, a sugestão é excluir o tarea. mas serão feitas exclusões após o stepwise.

### f)	Selecione um modelo pela técnica de stepwise;

In [215]:
resultados <- lm(price~age+parea+tarea+bath+ensuit+garag+plaz+park+trans+kidca+school+health+bike+barb+balc+elev+
                   fitg+party+categ,
                 data=imoveiscwbav)
summary (resultados)

library(RcmdrMisc)
step <- stepwise(resultados, direction= 'backward/forward', criterion ='AIC')
step


Call:
lm(formula = price ~ age + parea + tarea + bath + ensuit + garag + 
    plaz + park + trans + kidca + school + health + bike + barb + 
    balc + elev + fitg + party + categ, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-495718 -134211   -2632  104528 2419265 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -420453.5   130052.5  -3.233   0.0013 ** 
age           -7839.1     1025.3  -7.645 1.01e-13 ***
parea          2592.2      624.0   4.154 3.82e-05 ***
tarea          1975.8      333.9   5.918 5.91e-09 ***
bath          13452.6    14832.9   0.907   0.3649    
ensuit       125949.6    18560.7   6.786 3.15e-11 ***
garag        169687.5    21756.1   7.800 3.41e-14 ***
plaz         224393.0    94219.1   2.382   0.0176 *  
park         -63439.6    27154.0  -2.336   0.0199 *  
trans         26642.3    22718.5   1.173   0.2414    
kidca         10452.8    34899.8   0.300   0.7647    
school        -7975.8    56635.7  -0.141


Direction:  backward/forward
Criterion:  AIC 

Start:  AIC=13374.96
price ~ age + parea + tarea + bath + ensuit + garag + plaz + 
    park + trans + kidca + school + health + bike + barb + balc + 
    elev + fitg + party + categ

         Df  Sum of Sq        RSS   AIC
- health  1 2.4678e+07 2.7416e+13 13373
- school  1 1.0436e+09 2.7417e+13 13373
- kidca   1 4.7205e+09 2.7421e+13 13373
- bath    1 4.3284e+10 2.7459e+13 13374
- trans   1 7.2369e+10 2.7488e+13 13374
- party   1 8.6251e+10 2.7502e+13 13375
<none>                 2.7416e+13 13375
- bike    1 1.2339e+11 2.7539e+13 13375
- barb    1 1.9875e+11 2.7615e+13 13377
- park    1 2.8722e+11 2.7703e+13 13379
- plaz    1 2.9847e+11 2.7715e+13 13379
- balc    1 3.5048e+11 2.7767e+13 13380
- parea   1 9.0807e+11 2.8324e+13 13391
- fitg    1 9.8402e+11 2.8400e+13 13392
- elev    1 1.0269e+12 2.8443e+13 13393
- categ   1 1.3613e+12 2.8777e+13 13399
- tarea   1 1.8429e+12 2.9259e+13 13408
- ensuit  1 2.4231e+12 2.9839e+13 13419
- age    


Call:
lm(formula = price ~ age + parea + tarea + ensuit + garag + plaz + 
    park + trans + bike + barb + balc + elev + fitg + categ, 
    data = imoveiscwbav)

Coefficients:
(Intercept)          age        parea        tarea       ensuit        garag  
    -432184        -7608         2674         2048       135392       172755  
       plaz         park        trans         bike         barb         balc  
     227540       -62911        29844       -81354       -41485        74630  
       elev         fitg        categ  
     -98464       138618       295414  


##### Portanto o melhor modelo é:

In [230]:
resultados <- lm(price ~ age + parea + tarea + ensuit + garag + plaz + 
                   park + trans + bike + barb + balc + elev + fitg + categ, data = imoveiscwbav)
summary(resultados)


Call:
lm(formula = price ~ age + parea + tarea + ensuit + garag + plaz + 
    park + trans + bike + barb + balc + elev + fitg + categ, 
    data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-498873 -131074   -2422  106907 2398847 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -432184.4   105985.0  -4.078 5.25e-05 ***
age           -7608.4      998.3  -7.621 1.18e-13 ***
parea          2674.3      608.7   4.394 1.35e-05 ***
tarea          2047.9      329.0   6.225 9.83e-10 ***
ensuit       135391.5    15528.3   8.719  < 2e-16 ***
garag        172754.5    21324.7   8.101 3.83e-15 ***
plaz         227540.0    89152.2   2.552  0.01098 *  
park         -62910.6    22818.6  -2.757  0.00604 ** 
trans         29843.6    20366.1   1.465  0.14342    
bike         -81354.4    52369.0  -1.553  0.12091    
barb         -41484.8    22232.6  -1.866  0.06261 .  
balc          74630.0    24206.1   3.083  0.00216 ** 
elev         -98463.8    23

### g)	Faça o teste de homocedasticidade e faça correção da heterocedasticidade se necessário;

In [231]:
bptest(price ~ age + parea + tarea + ensuit + garag + plaz + 
         park + trans + bike + barb + balc + elev + fitg + categ,studentize=FALSE, data=imoveiscwbav)


	Breusch-Pagan test

data:  price ~ age + parea + tarea + ensuit + garag + plaz + park +     trans + bike + barb + balc + elev + fitg + categ
BP = 290, df = 14, p-value < 2.2e-16


##### Obtendo o valor quiquadrado tabelado

In [232]:
qchisq(0.95, df=14, lower.tail = TRUE)

[1] 23.68479

O valor de 290 está muito acima do tabelado, então rejeita-se a hipótese de hocedascidade. Será feito regressão robusta.

In [234]:
library(robust)

resultrob <- lmRob(price ~ age + parea + tarea + ensuit + garag + plaz + 
                     park + trans + bike + barb + balc + elev + fitg + categ,data=imoveiscwbav)
summary(resultrob)

00:00:01 left
00:00:01 left
00:00:01 left
00:00:01 left
00:00:01 left
00:00:01 left
00:00:01 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left


Warning message in lmRob.fit.compute(x, y, x1.idx = x1.idx, nrep = nrep, robust.control = robust.control, :
"Max iteration for refinement reached."


Call:
lmRob(formula = price ~ age + parea + tarea + ensuit + garag + 
    plaz + park + trans + bike + barb + balc + elev + fitg + 
    categ, data = imoveiscwbav)

Residuals:
    Min      1Q  Median      3Q     Max 
-479369  -97422    4898  115760 2507441 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -483934.3    83244.9  -5.813 1.06e-08 ***
age           -7293.5      741.1  -9.841  < 2e-16 ***
parea          2681.1      483.5   5.545 4.66e-08 ***
tarea          1549.4      284.9   5.439 8.22e-08 ***
ensuit       134199.9    13209.0  10.160  < 2e-16 ***
garag        146425.7    16707.0   8.764  < 2e-16 ***
plaz         321773.4    67936.3   4.736 2.80e-06 ***
park         -32549.7    17591.4  -1.850  0.06483 .  
trans         36201.5    15078.7   2.401  0.01670 *  
bike         -97602.6    40426.2  -2.414  0.01610 *  
barb          -7928.0    17619.6  -0.450  0.65293    
balc          54352.5    18933.1   2.871  0.00426 ** 
elev         -64990.3   

### h)	Obtenha os indicadores de desempenho do modelo;

In [235]:
AIC(resultados)
BIC(resultados)

[1] 14905.22

[1] 14973.92

In [236]:
library(AICcmodavg)
AICc(resultados)

[1] 14906.26

In [173]:
library(performance)
model_performance(resultados)
model_performance(resultrob)

AIC      BIC      R2        R2_adjusted RMSE     Sigma   
1 14905.22 14973.92 0.8088607 0.8037734   225733.1 228929.1

R2        RMSE     Sigma 
1 0.7905655 236289.5 239635

### i)	Estime os intervalos de confiança para os parâmetros do modelo;

In [237]:
# Calculando os intervalos de confiança - Para regressão linear "normal"

confint(resultados, level = 0.95)

2.5 %       97.5 %     
(Intercept) -640390.197 -223978.663
age           -9569.572   -5647.220
parea          1478.579    3869.970
tarea          1401.687    2694.174
ensuit       104886.352  165896.726
garag        130862.467  214646.556
plaz          52401.923  402678.148
park        -107737.383  -18083.784
trans        -10165.240   69852.514
bike        -184232.529   21523.711
barb         -85160.380    2190.798
balc          27077.574  122182.509
elev        -144760.587  -52166.944
fitg          88316.975  188919.288
categ        190282.870  400545.700

In [238]:
# Calculando os intervalos de confiança - Para a regressão robusta

confint(resultrob, level=0.95)

2.5 %        97.5 %     
(Intercept) -647091.1924 -320777.322
age           -8746.0011   -5840.952
parea          1733.4274    3628.755
tarea           991.0716    2107.736
ensuit       108310.7009  160089.136
garag        113680.5160  179170.787
plaz         188620.6910  454926.198
park         -67028.1510    1928.786
trans          6647.7359   65755.211
bike        -176836.4768  -18368.795
barb         -42461.7478   26605.673
balc          17244.2960   91460.756
elev        -102680.7428  -27299.923
fitg          60802.5216  140534.528
categ        289785.2257  452885.540

### j)	Faça predição de um imóvel hipotético: apresente seus parâmetros de simulação e o resultado.

In [239]:
#age
log(5)

[1] 1.609438

In [240]:
#educ
log(150)

[1] 5.010635

In [241]:
#earns
log(190)

[1] 5.247024

In [242]:
# plaz
log(0.08)

[1] -2.525729

In [243]:
#parea
log(150)

[1] 5.010635

In [244]:
#tarea
log(190)

[1] 5.247024

In [249]:
#kidlt6 = 1 --> tem filhos com menos de 6 anos
#kidge6 =0 --> parâmetro não significativo a 95%
#union = 0  --> parâmetro não significativo a 95%
#husearns = 0 --> parâmetro não significativo a 95%
#hushrs = 0 --> parâmetro não significativo

predict(object = resultrob,
        data.frame(age=1.60, parea=5.01, tarea=5.24, ensuit = 1, garag = 1, plaz = -2.525729, 
                   park = -2.525729, trans = -2.525729, bike = -2.525729, categ = 1,
                   barb=0, balc=0, elev=0, fitg=0))

1 
-397509.9

In [255]:
exp(-397509.870811602)

[1] 0

In [256]:
anova.lmRob(resultrob)
anova(resultados)

Warning message in lmRob.fit.compute(x, y, x1.idx = x1.idx, nrep = nrep, robust.control = robust.control, :
"Max iteration for refinement reached."Warning message in lmRob.fit.compute(x, y, x1.idx = x1.idx, nrep = nrep, robust.control = robust.control, :
"Max iteration for refinement reached."

00:00:00 left


Warning message in lmRob.fit.compute(x, y, x1.idx = x1.idx, nrep = nrep, robust.control = robust.control, :
"Max iteration for refinement reached."

00:00:00 left
00:00:00 left
00:00:00 left


Warning message in lmRob.fit.compute(x, y, x1.idx = x1.idx, nrep = nrep, robust.control = robust.control, :
"Max iteration for refinement reached."

00:00:01 left
00:00:01 left
00:00:01 left
00:00:00 left
00:00:00 left
00:00:00 left
00:00:00 left


Chisq Df RobustF      Pr(F)       
(Intercept) 1                  NA           NA
age         1        205.08139660 0.000000e+00
parea       1        410.02790435 0.000000e+00
tarea       1         84.91062412 0.000000e+00
ensuit      1         77.36769245 0.000000e+00
garag       1         58.07340719 8.104628e-15
plaz        1         11.58140111 5.245293e-04
park        1         14.02048281 1.358258e-04
trans       1          1.72004165 1.813986e-01
bike        1         12.32212109 3.474436e-04
barb        1          0.06918982 7.886646e-01
balc        1          5.49291547 1.692666e-02
elev        1          5.11284106 2.121207e-02
fitg        1         28.85005407 4.413883e-08
categ       1         55.34172989 3.430589e-14

Df  Sum Sq       Mean Sq      F value     Pr(>F)       
age         1 4.118120e+13 4.118120e+13 785.7727779 1.881083e-106
parea       1 4.962541e+13 4.962541e+13 946.8954239 1.063549e-119
tarea       1 7.586388e+12 7.586388e+12 144.7548093  1.271275e-29
ensuit      1 6.588508e+12 6.588508e+12 125.7144024  2.614428e-26
garag       1 4.407336e+12 4.407336e+12  84.0957552  1.062007e-18
plaz        1 7.644615e+10 7.644615e+10   1.4586583  2.276865e-01
park        1 2.016816e+12 2.016816e+12  38.4825855  1.119422e-09
trans       1 2.261392e+10 2.261392e+10   0.4314930  5.115445e-01
bike        1 2.333090e+11 2.333090e+11   4.4517355  3.533747e-02
barb        1 3.777160e+10 3.777160e+10   0.7207145  3.962958e-01
balc        1 9.103650e+11 9.103650e+11  17.3705471  3.595692e-05
elev        1 3.750541e+11 3.750541e+11   7.1563540  7.702535e-03
fitg        1 1.999045e+12 1.999045e+12  38.1434931  1.316225e-09
categ       1 1.596971e+12 1.596971e+12  30.4715859  5.327990e-08
Residuals 526 2.756689e+13 5.240854e+10          NA            NA

#### Nota:
<p>Resultados foram Heterogêneos e inconsistentes. Infelizmente não fui capaz de tratar as informações para atingir um resultado satisfatório. Estou estudando para ver se minha dificuldade está nos dados, nos métodos ou algum conflito de bibliotecas das ferramentas que estou usando.</p>